lib/odometry-ng/include/odometer/OdometerDifferential.hpp

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#include <cstdint>
+
+#include "encoder/EncoderInterface.hpp"
+#include "OdometerInterface.hpp"
+#include "OdometerDifferentialParameters.hpp"
+
+namespace cogip {
+
+namespace odometer {
+
+class OdometerDifferential: public OdometerInterface {
+
+public:
+    /// @brief Construct a new differential odometer object
+    /// @param parameters Odometer parameters
+    explicit OdometerDifferential(OdometerDifferentialParameters &parameters) : parameters_(parameters), left_encoder_(NULL), right_encoder_(NULL) {}
```

Comment:
s/NULL/nullptr

But i think encoders should be passed in constructor and be marked as const

Reply:
The encoder cannot be const because there are called inside the update method that need to modify call members to compute the position and velocity.

But I now pass the constructor reference on the constructor to avoid pointers.